\documentclass[12pt]{article}
\usepackage{lecture}
\usepackage{html}
\usepackage{graphics}
\usepackage{epstopdf}

\newcommand{\copyrightYears}{2004-2015}

\title{Analyzing the genetic structure of populations: individual assignment}

\begin{document}

\maketitle

\thispagestyle{first}

\section*{Introduction}

About 15 years ago a different approach to the analysis of genetic
structure began to emerged: analysis of individual
assignment.\index{individual assignment} Although the implementation
details get a little hairy, the basic idea is fairly simple. Suppose
we have genetic data on a series of individuals. Label the data we
have for each individual $x_i$. Suppose that all individuals belong to
one of $K$ populations and let the genotype frequencies in population
$k$ be represented by $\gamma_k$. Then the likelihood that individual
$i$ comes from population $k$ is just
\[
\mbox{P}(i|k) = \frac{\mbox{P}(x_i|\gamma_k)}{\sum_k
  \mbox{P}(x_i|\gamma_k)} \quad .
\]
So if we can specify prior probabilities for $\gamma_k$, we can use
Bayesian methods to estimate the posterior probability that individual
$i$ belongs to population $k$, and we can associate that assignment
with some measure of its reliability.\footnote{You can find details
  in~\cite{Pritchard-etal-2000}.}

\section*{Applying assignment to understand invasions}

We'll use {\tt Structure} to assess whether cultivated genotypes of
{\it Berberis thunbergii\/} contribute to ongoing invasions in
Connecticut and
Massachusetts~\cite{Lubell-etal-2008}.\index{individual assignment!application} The first problem is to determine what $K$
to use, because $K$ doesn't necessarily have to equal the number of
populations we sample from. Some populations may not be distinct from
one another. There are a couple of ways to estimate $K$. The most
straightforward is to run the analysis for a range of plausible
values, repeat it 10-20 times for each value, calculate the mean ``log
probability of the data'' for each value of $K$, and pick the value of
$K$ that is the biggest, i.e., the least
negative~(Table~\ref{table:berberis-k}). For the barberry data, $K=3$
is the obvious choice.

\begin{table}
\begin{center}
\begin{tabular}{cc}
\hline\hline
K & Mean L(K) \\
\hline
2 & -2553.2 \\
3 & {\bf -2331.9} \\
4 & -2402.9 \\
5 & -2476.3 \\
\hline
\end{tabular}
\end{center}
\caption{Mean log probability of the data for $K=2,3,4,5$ in the {\it
    Berberis thunbergii\/} data~(adapted
  from~\cite{Lubell-etal-2008}).}\label{table:berberis-k}
\end{table}

Having determined that the data support $K=3$, the results of the
analysis are displayed in Figure~\ref{fig:lubell-structure}. Each
vertical bar corresponds to an individual in the sample, and the
proportion of each bar that is of a particular color tells us the
posterior probability that the individual belongs to the cluster with
that color.

\begin{figure}
\resizebox{\textwidth}{!}{\includegraphics{lubell-structure.eps}}
\caption{Analysis of AFLP data from {\it Berberis
    thunbergii}~\cite{Lubell-etal-2008}.}\label{fig:lubell-structure} 
\end{figure}

Figure~\ref{fig:lubell-structure} may not look terribly informative,
but actually it is. Look at the labels beneath the figure. You'll see
that with te exception of individual 17 from Beaver Brook Park, all
the of the individuals that are solid blue are members of the
cultivated {\it Berberis thunbergii\/} var. {\it atropurpurea}. The
solid red bar corresponds to {\it Berberis thunbergii\/}
'Atropurpurea', another modern cultivar. You'll notice that
individuals 1, 2, 18, and 19 from Beaver Brook Park and individual 1
from Bluff Point State Park fall into the same genotypic cluster as
this cultivar. {\it Berberis $\times$ottawensis} is a hybrid cultivar
whose parents are {\it Berberis thunbergii\/} and {\it Berberis
  vulgaris\/}, so it makes sense that individuals of this cultivar
would be half blue and half red. The solid green bars are feral
individuals from long-established populations. Notice that the
cultivars are distinct from all but a few of the individuals in the
long-established feral populations, suggesting that contemporary
cultivars are doing relatively little to maintain the invasion in
areas where it is already established.

\section*{Genetic diversity in human populations}\index{humans}

A much more interesting application of {\tt Structure} appeared a
little over a decade ago. The Human Genome Diversity Cell Line Panel
(HGDP-CEPH)\index{HGDP-CEPH} consisted at the time of data from 1056
individuals in 52 geographic populations. Each individual was
genotyped at 377 autosomal loci. If those populations are grouped into
5 broad geographical regions~(Africa, [Europe, the Middle East, and
Central/South Asia], East Asia, Oceania, and the Americas), we find
that about 93\% of genetic variation is found within local populations
and only about 4\% is a result of allele frequency differences between
regions~\cite{Rosenberg-etal-2002}. You might wonder why Europe, the
Middle East, and Central/South Asia were grouped together for that
analysis. The reason becomes clearer when you look at a {\tt
  Structure} analysis of the same data~(Figure~\ref{fig:HGDP-CEPH}).

\begin{figure}
\resizebox{\textwidth}{!}{\includegraphics{HGDP-CEPH.eps}}
\caption{{\tt Structure} analysis of microsatellite diversity in the
  Human Genome Diversity Cell Line
  Panel~(from~\cite{Rosenberg-etal-2002}).}\label{fig:HGDP-CEPH} 
\end{figure}

\subsection*{A non-Bayesian look at individual-based analysis of
  genetic structure}

{\tt Structure} has a lot of nice features, but you'll discover a
couple of things about it if you begin to use it seriously: (1) It
often isn't obvious what the ``right'' {\tt K} is.\footnote{In fact,
  it's not clear that there {\it is\/} such a thing as the ``right''
  {\tt K}. If you're interested in hearing more about that. Feel free
  to ask.} (2) It requires a {\it lot\/} of computational resources,
especially with datasets that include a few thousand SNPs, as is
becoming increasingly common. An alternative is to use principal
component analysis directly on genotypes. There are technical details
associated with estimating the principal components and interpreting
them that we won't discuss,\footnote{See~\cite{Novembre-Stephens-2008}
  for details}, but the results can be pretty
striking. Figure~\ref{fig:human-PCA} shows the results of a PCA on
data derived from 3192 Europeans at 500,568 SNP loci. The
correspondence between the position of individuals in PCA space and
geographical space is remarkable.

\begin{figure}
\resizebox{\textwidth}{!}{\includegraphics{human-PCA.eps}}
\caption{Principal components analysis of genetic diversity in Europe
  corresponds with
  geography~(from~\cite{Novembre-etal-2008}). Panel b is a close-up
  view of the area around Switzerland~(CH).}\label{fig:human-PCA}
\end{figure}

\bibliography{popgen}
\bibliographystyle{plain}

\ccLicense

\end{document}
